src/NosCore.Controllers/Inventory/UpgradePacketHandler.cs

```diff
@@ -0,0 +1,94 @@
+﻿using ChickenAPI.Packets.ClientPackets.Player;
+using ChickenAPI.Packets.Enumerations;
+using NosCore.Core.I18N;
+using NosCore.Data;
+using NosCore.Data.Enumerations.I18N;
+using NosCore.GameObject;
+using NosCore.GameObject.Networking.ClientSession;
+using NosCore.GameObject.Providers.ItemProvider.Item;
+using Serilog;
+
+namespace NosCore.PacketHandlers.Inventory
+{
+    public class UpgradePacketHandler : PacketHandler<UpgradePacket>, IWorldPacketHandler
+    {
+        private readonly ILogger _logger;
+
+        public UpgradePacketHandler(ILogger logger)
+        {
+            _logger = logger;
+        }
+
+        public override void Execute(UpgradePacket packet, ClientSession clientSession)
+        {
+            switch (packet.UpgradeType)
+            {
+                case UpgradePacketType.ItemToPart:
+                    break;
+                case UpgradePacketType.UpgradeItem:
+                    break;
+                case UpgradePacketType.CellonItem:
+                    break;
+                case UpgradePacketType.RarifyItem:
+                    break;
+                case UpgradePacketType.SumResistance:
```

Comment:
Cant really accept a huge switch like that. Would be nice to have a UpgradeService

Reply:
Or maybe something event based such as the n_run system, that would make sense imo

Reply:
this is what i meant by service.